src/backend/tools.ts

```diff
@@ -146,51 +181,151 @@ export const DXVK = {
         .split('\n')[0]
     }
 
-    const installCommand = `PATH=${wineBin}:$PATH WINEPREFIX='${winePrefix}' bash ${toolPath}/setup*.sh install --symlink`
-
     if (action === 'restore') {
       logInfo(`Removing ${tool} version information`, {
         prefix: LogPrefix.DXVKInstaller
       })
-      const updatedVersionfile = `rm -rf ${currentVersionCheck}`
-      const removeCommand = `PATH=${wineBin}:$PATH WINEPREFIX='${winePrefix}' bash ${toolPath}/setup*.sh uninstall --symlink`
-      return execAsync(removeCommand, execOptions)
-        .then(() => {
-          logInfo(`${tool} removed from ${winePrefix}`, {
-            prefix: LogPrefix.DXVKInstaller
-          })
-          return exec(updatedVersionfile)
+      if (existsSync(currentVersionCheck)) {
+        rmSync(currentVersionCheck)
+      }
+
+      logInfo(`Removing ${tool} files`, {
+        prefix: LogPrefix.DXVKInstaller
+      })
+
+      // remove the dlls from the prefix
+      await new Promise((resolve) => {
+        dlls.forEach((dll) => {
+          const dllPath = `${winePrefix}/drive_c/windows/system32/${dll}`
+          if (existsSync(`${dllPath}.bak`)) {
+            const removeDll = `rm ${dllPath}`
+            exec(removeDll)
+          }
+
+          const dllPath64 = `${winePrefix}/drive_c/windows/syswow64/${dll}`
+          if (existsSync(`${dllPath}.bak`)) {
+            const removeDll = `rm ${dllPath64}`
+            exec(removeDll)
+          }
         })
-        .catch((error) => {
-          logError(['error when removing DXVK, please try again', error], {
-            prefix: LogPrefix.DXVKInstaller
-          })
+        resolve(true)
+      })
+
+      // restore the backup dlls
+      await new Promise((resolve) => {
+        dlls.forEach((dll) => {
+          const dllPath = `${winePrefix}/drive_c/windows/system32/${dll}.bak`
+          const dllPath64 = `${winePrefix}/drive_c/windows/syswow64/${dll}.bak`
+          logInfo(`Restoring ${dll}`, { prefix: LogPrefix.DXVKInstaller })
+          if (existsSync(dllPath)) {
+            renameSync(dllPath, `${winePrefix}/drive_c/windows/system32/${dll}`)
+          }
+
+          if (!isMac) {
+            if (existsSync(dllPath64)) {
+              renameSync(
+                dllPath64,
+                `${winePrefix}/drive_c/windows/syswow64/${dll}`
+              )
+            }
+          }
+        })
+        resolve(true)
+      })
+
+      // unregister the dlls on the wine prefix
+      await new Promise((resolve) => {
+        dlls.forEach((dll) => {
+          const unregisterDll = `WINEPREFIX='${winePrefix}' '${wineBin}' reg delete 'HKEY_CURRENT_USER\\Software\\Wine\\DllOverrides' /v ${dll} /f`
+          logInfo(`Unregistering ${dll}`, { prefix: LogPrefix.DXVKInstaller })
+          exec(unregisterDll)
         })
+        resolve(true)
+      })
```

Comment:
Are all of these `await new Promise`s necessary? Why not just do this directly?
```suggestion
    dlls.forEach((dll) => {
        const unregisterDll = `WINEPREFIX='${winePrefix}' '${wineBin}' reg delete 'HKEY_CURRENT_USER\\Software\\Wine\\DllOverrides' /v ${dll} /f`
        logInfo(`Unregistering ${dll}`, { prefix: LogPrefix.DXVKInstaller })
        exec(unregisterDll)
    })
```
`exec` will not block regardless, if that's what you wanted to do. For that to work, you'd need a callback:
```ts
exec(unregisterDll, undefined, () => resolve(true))
```
(or you just use `execAsync` & `await` it)

Reply:
We need the await to avoid the racing condition. We can start copying before backing up, so we will overwrite it. This happened during my tests.

The registering is not the issue, we can register the DLLs even if they do not exist in the path, or of they are the wrong ones. The problem is only with backing up them. But I wanted to do it in the right order so on the logs we can have the correct execution.

Reply:
Right, but this does not change this. The race condition is caused by `exec` not blocking, and just putting it in a Promise won't change that (at least if you're doing it like that)

A simple example:
```ts
import { exec } from 'child_process'

await new Promise<void>((res) => {
  exec('sleep 5', undefined, () => console.log('Process done'))
  res()
})
console.log('Promise resolved')
```
Output:
```
$ ts-node-esm index.ts
Promise resolved
Process done
```

If you want to wait for the process to finish, either resolve the Promise in the callback function of `exec`, or just use `execAsync`

Reply:
Yes, funny thing is that I did use `execAsync` when registering but when registering I used `exec`.
I should stop programming 18h a day 🤦🏽

Reply:
It's impossible to know/remember everything, especially about a language as vast as JS/TS. That's why reviews exist, after all